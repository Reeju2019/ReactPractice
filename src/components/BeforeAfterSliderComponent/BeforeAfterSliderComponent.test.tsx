import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import { BeforeAfterSliderComponent } from "."

const beforeAfterData = {
  image1: {
    id: 1,
    img: "https://media.geeksforgeeks.org/wp-content/uploads/20210425122739/2-300x115.png",
    alt: "Image One label",
    header: "First slide label",
  },
  image2: {
    id: 2,
    img: "https://images.news18.com/ibnlive/uploads/2021/08/1628056310_dogdrinking-1200x800.png?impolicy=website&width=510&height=356",
    alt: "Image two label",
    header: "Second slide label",
  },
}

/**
 * Unit testing for BeforeAfterSlider Component
 */
describe("Before after Slider Component Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <BeforeAfterSliderComponent
        aspectRatio="taller"
        data={beforeAfterData}
      />,
      div
    )
    // ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <BeforeAfterSliderComponent aspectRatio="taller" data={beforeAfterData} />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
