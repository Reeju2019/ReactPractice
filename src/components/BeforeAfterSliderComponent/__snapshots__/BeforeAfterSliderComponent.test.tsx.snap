// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Before after Slider Component Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Row
      className="container m-3 p-2"
      md={12}
      sm={12}
    >
      <s
        aspectRatio="taller"
        handle={null}
        handleSize={40}
        hover={false}
        leftImage="https://media.geeksforgeeks.org/wp-content/uploads/20210425122739/2-300x115.png"
        leftImageAlt="Image One label"
        leftImageCss={Object {}}
        leftImageLabel="First slide label"
        onSliderPositionChange={null}
        rightImage="https://images.news18.com/ibnlive/uploads/2021/08/1628056310_dogdrinking-1200x800.png?impolicy=website&width=510&height=356"
        rightImageAlt="Image two label"
        rightImageCss={Object {}}
        rightImageLabel="Second slide label"
        skeleton={null}
        sliderLineColor="#ffffff"
        sliderLineWidth={2}
        sliderPositionPercentage={0.5}
        vertical={false}
      />
    </Row>
  </React.Fragment>,
]
`;
